### PR TITLE
Factory method for Grid

### LINKED ISSUE #213: Add Grid.from_config() factory function
**Is your feature request related to a problem? Please describe.**

As noted:

https://github.com/ImperialCollegeLondon/virtual_rainforest/blob/dd258ef988e3c14fae10db1456d57a2753e95f24/virtual_rainforest/main.py#L207

There is currently no way to generate a `Grid` object from a configuration. And there needs to be.

**Describe the solution you'd like**
Write the `Grid.from_config()` method.

### PR DESCRIPTION
**Pull request by davidorme**
# Description

This PR:
* Introduces the `Grid.from_config` factory method.
* Updates the core schema to include all of the arguments to `Grid.__init__` to allow full configuration.
* Updates the names of the `core.grid.nx` and `core.grid.ny` to `cell_n...` to match the `Grid.__init__` arguments.
* Adds a test for the `Grid.from_config` method.
* Updates the rest of the test suite for the schema name changes.

Fixes #213 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] Make sure you've run the `pre-commit` checks: `$ pre-commit run -a`
- [x] All tests pass: `$ poetry run pytest`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: virtual_rainforest/core/core_schema.json [hunk 1]
```diff
@@ -9,23 +9,44 @@
...
                "type": "object",
                "properties": {
-                  "nx": {
+                  "grid_type": {
+                     "description": "The grid cell type",
+                     "type": "string",
+                     "default": "square"
+                  },
+                  "cell_area": {
+                     "description": "The area of each grid cell",
```
Comment (jacobcook1995) [inline virtual_rainforest/core/core_schema.json:18]: Is it worth mentioning a unit here?

File: virtual_rainforest/core/core_schema.json [hunk 2]
```diff
@@ -9,23 +9,44 @@
...
+                     "description": "The y offset of the grid origin",
+                     "type": "number",
+                     "default": 0
                   }
                },
                "default": {},
                "required": [
-                  "nx",
-                  "ny"
+                  "cell_nx",
```
Comment (jacobcook1995) [inline virtual_rainforest/core/core_schema.json:48]: Not really an important point, but I've just realised that `cell_nx` and `cell_ny` don't need to be required as they have defaults set. I think we could just have `"required": []` here.

**Review by jacobcook1995 — APPROVED** [review]
LGTM! Couple of minor points but nothing major

File: virtual_rainforest/core/core_schema.json (hunk 1 above)
Comment (davidorme) [reply virtual_rainforest/core/core_schema.json:18]: Yup - also added some clarification in `core.grid` docstrings

File: virtual_rainforest/core/core_schema.json (hunk 2 above)
Comment (davidorme) [reply virtual_rainforest/core/core_schema.json]: I'd thought the same - have updated.

**Review by jacobcook1995 — APPROVED** [review]
LGTM!

**Pull request merged**